packages/grid/test/styling.common.js

```diff
@@ -197,4 +197,91 @@ describe('styling', () => {
 
     runStylingTest('parts', 'cellPartNameGenerator', 'generateCellPartNames', assertPartNames);
   });
+
+  describe('header and footer part name', () => {
+    let column;
+    let headerCell;
+    let footerCell;
+
+    beforeEach(() => {
+      column = grid.querySelector('vaadin-grid-column');
+      column.footerRenderer = (root) => {
+        root.textContent = 'footer';
+      };
+      headerCell = getContainerCell(grid.$.header, 0, 0);
+      footerCell = getContainerCell(grid.$.footer, 0, 0);
+    });
+
+    it('should add a header and footer part name', () => {
+      column.headerPartName = 'foobar';
+      column.footerPartName = 'bazqux';
+
+      expect(headerCell.getAttribute('part')).to.contain('foobar');
+      expect(footerCell.getAttribute('part')).to.contain('bazqux');
```

Comment:
nit: just to ensure we are not messing up anything here, could it also check that the part names are not mistakenly added to the wrong cell? Something like:

```suggestion
      expect(headerCell.getAttribute('part')).to.contain('foobar');
      expect(headerCell.getAttribute('part')).to.not.contain('bazqux');
      expect(footerCell.getAttribute('part')).to.contain('bazqux');
      expect(footerCell.getAttribute('part')).to.not.contain('foobar');
```

Reply:
I think this test already covers it https://github.com/vaadin/web-components/pull/6629/files#diff-1efbbf12b5ed1e54feef4de97f7612f5b5f335077eaf3262d3ef758f02a02615R248-R254

Reply:
Well, sort of. NVM then.